rsciio/utils/distributed.py

```diff
@@ -0,0 +1,169 @@
+    block_size_limit=None,
+    dtype=None,
+):
+    """
+    Get chunk slices for the :func:`rsciio.utils.distributed.slice_memmap` function.
+
+    Takes a shape and chunks and returns a dask array of the slices to be used with the
+    :func:`rsciio.utils.distributed.slice_memmap` function. This is useful for loading data from a memmaped file in a
+    distributed manner.
+
+    Parameters
+    ----------
+    shape : tuple
+        Shape of the data.
+    chunks : tuple, optional
+        Chunk shape. The default is "auto".
+    block_size_limit : int, optional
+        Maximum size of a block in bytes. The default is None.
+    dtype : numpy.dtype, optional
+        Data type. The default is None.
+
+    Returns
+    -------
+    dask.array.Array
+        Dask array of the slices.
+    """
+
+    chunks = da.core.normalize_chunks(
+        chunks=chunks, shape=shape, limit=block_size_limit, dtype=dtype
+    )
+    chunks_shape = tuple([len(c) for c in chunks])
+    slices = np.empty(
+        shape=chunks_shape
+        + (
+            len(chunks_shape),
+            2,
+        ),
+        dtype=int,
+    )
+    for ind in np.ndindex(chunks_shape):
+        current_chunk = [chunk[i] for i, chunk in zip(ind, chunks)]
+        starts = [int(np.sum(chunk[:i])) for i, chunk in zip(ind, chunks)]
+        stops = [s + c for s, c in zip(starts, current_chunk)]
+        slices[ind] = [[start, stop] for start, stop in zip(starts, stops)]
+    return da.from_array(slices, chunks=(1,) * len(shape) + slices.shape[-2:]), chunks
+
+
+def slice_memmap(sl, file, dtypes, shape, **kwargs):
+    """
+    Slice a memmaped file using a tuple of slices.
+
+    This is useful for loading data from a memmaped file in a distributed manner. This takes
+    a slice of the dimensions of the data and returns the data from the memmaped file sliced.
+
+    Parameters
+    ----------
+    sl : array-like
+        An array of the slices to use. The dimensions of the array should be
+        (n,2) where n is the number of dimensions of the data. The first column
+        is the start of the slice and the second column is the stop of the slice.
+    file : str
+        Path to the file.
+    dtypes : numpy.dtype
+        Data type of the data for memmap function.
+    shape : tuple
+        Shape of the data to be read.
+    **kwargs : dict
+        Additional keyword arguments to pass to the memmap function.
+
+    Returns
+    -------
+    numpy.ndarray
+        Array of the data from the memmaped file sliced using the provided slice.
+    """
+    sl = np.squeeze(sl)[()]
+    data = np.memmap(file, dtypes, shape=shape, **kwargs)
+    slics = tuple([slice(s[0], s[1]) for s in sl])
+    return data[slics]
+
+
+def memmap_distributed(
+    file,
+    dtype,
+    offset=0,
+    shape=None,
+    order="C",
+    chunks="auto",
+    block_size_limit=None,
+):
+    """
+    Drop in replacement for py:func:`numpy.memmap` allowing for distributed loading of data.
```

Comment:
```suggestion
    Drop in replacement for :class:`numpy.memmap` allowing for distributed loading of data.
```

Reply:
Huh.. I didn't realize that was a class.  Numpy gets a little weird with not capitalizing their classes...